mlflow/pipelines/utils/step.py

```diff
@@ -122,10 +122,26 @@ def get_pandas_data_profile(data_frame, title: str):
 
     max_cells = min(data_frame.size, _MAX_PROFILE_CELL_SIZE)
     max_cols = min(data_frame.columns.size, _MAX_PROFILE_COL_SIZE)
-    max_rows = min(max(max_cells // max_cols, 1), _MAX_PROFILE_ROW_SIZE)
+    max_rows = min(
+        max(max_cells // max_cols, 1),
+        (data_frame.size // data_frame.columns.size),
+        _MAX_PROFILE_ROW_SIZE,
+    )
```

Comment:
As proposed offline, I think it may be simpler for us to skip sampling if `len(df) < max_rows`.

Reply:
Spoke offline that this might be a bit easier rather than adding a if check:
```
   max_rows = min(
        max(max_cells // max_cols, 1),
        len(data_frame),
        _MAX_PROFILE_ROW_SIZE,
    )
 ```